src/scripts/utils.py

```diff
@@ -0,0 +1,399 @@
+    #     # Get params from wherever supplied
+    #     prefix_map = owned_prefix_map if owned_prefix_map else self.owned_prefix_map
+    #     onto = ontology if ontology else self.ontology
+    #     if not (prefix_map and onto):
+    #         raise ValueError('Term._get_direct_owned_parents(): Must supply both `ontology` and `owned_prefix_map`.')
+    #
+    #     self.direct_owned_parent_uris = _get_direct_owned_parent_uris(
+    #         curie=self.curie, ontology=self.ontology, owned_prefix_map=self.owned_prefix_map)
+    #     if self.direct_owned_parent_uris:
+    #         self.direct_owned_parent_curies = [self._converter.compress(x) for x in self.direct_owned_parent_uris]
+    #     return self.direct_owned_parent_uris
+
+    def fetch_missing_props(
+        self, ontology: ProntoImplementation = None, general_prefix_map: Dict[PREFIX, URI] = None,
+        owned_prefix_map: Dict[PREFIX, URI] = None
+    ):
+        """For any undefined properties, try to set their values using available params.
+
+        ontology: Requires use of `oaklib`, particularly the `ProntoImplementation`.
+        owned_prefix_map: A prefix_map that includes *only* of the prefixes 'owned' by the ontology. Used to
+        general_prefix_map: A general prefix_map. For conversion of URIs and CURIEs, even if not owned by ontology."""
+        # Get params from wherever supplied
+        ontology = ontology if ontology else self.ontology
+        general_prefix_map = general_prefix_map if general_prefix_map else self.general_prefix_map
+        owned_prefix_map = owned_prefix_map if owned_prefix_map else self.owned_prefix_map
+        if not (ontology or general_prefix_map or owned_prefix_map):
+            raise ValueError('Term.fetch_missing_props(): Must supply at least one param.')
+
+        # Set: uri, curie
+        prefix_map = general_prefix_map if general_prefix_map else owned_prefix_map
+        if prefix_map and self.uri and not self.curie:
+            converter = self._converter if self._converter else curies.Converter.from_prefix_map(prefix_map)
+            curie = converter.compress(self.uri)
+            self.curie = curie if curie != self.uri else None
+        elif prefix_map and self.curie and not self.uri:
+            converter = curies.Converter.from_prefix_map(prefix_map)
+            uri = converter.expand(self.curie)
+            self.uri = uri if uri != self.curie else None
+
+        # Set: label, definition
+        # todo: Assign in batch (when OAK supports): I feel like would be faster to do outside the Term class itself
+        #  ...e.g. build a dict of CURIE-> Term, then use use ontology.labels() on all curies and set label
+        #  ...Right now, OAK ProntoImplementation labels() is just proxy for label(), though SqlImplementation has, and
+        #  ...Both of them have definition() but not definitions().
+        if ontology:
+            self.label = ontology.label(self.uri)
```

Comment:
@matentzn **Fixed 'labels' issue**
OAK was not returning labels for some ontology. Seems to be a `prefix_map` issue. Also, the type definitions for these `.label()` functions only say they support CURIE, but they also support URI, which is how I was able to fix this bug. I updated the [corresponding OAK issue](https://github.com/INCATools/ontology-access-kit/issues/212) with these new updates.